macros/upload_results.sql

```diff
@@ -78,14 +78,23 @@
 
         {% do log("Uploading tests", true) %}
         {% set tests = dbt_artifacts.get_relation('tests') %}
-        {% set content_tests = dbt_artifacts.upload_tests(graph) %}
-        {{ dbt_artifacts.insert_into_metadata_table(
-            database_name=tests.database,
-            schema_name=tests.schema,
-            table_name=tests.identifier,
-            content=content_tests
-            )
-        }}
+        {% set tests_set = [] %}
+        {% for node in graph.nodes.values() | selectattr("resource_type", "equalto", "test") %}
+            {% do tests_set.append(node) %}
+        {% endfor %}
+        {# upload tests in chunks of 5000 tests, or less #}
+        {% set upload_limit = 5000 %}
```

Comment:
This looks like it would also be a great solution for #232 although BigQuery typically has a lower limit. So I wonder if it would be good to include something like this here:

```suggestion
        {% set upload_limit = 750 if target.type == 'bigquery' else 5000 %}
```